### PR TITLE
[Documentation] tarchiveLoader

### PR DESCRIPTION
**Pull request by MounaSafiHarab**
Generated using:
`pod2markdown uploadNeuroDB/tarchiveLoader > docs/scripts_md/tarchiveLoader.md`

User can type:
`perldoc uploadNeuroDB/tarchiveLoader`
to get the documentation at the terminal

### CODE REVIEW

**Comment by MounaSafiHarab** [comment]
discussion required because I am proposing adding the `.pl` extension to the filename.

**Comment by MounaSafiHarab** [comment]
last one for before the holidays, I will resume the rest after!

File: docs/scripts_md/tarchiveLoader.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# NAME
+
+tarchiveLoader.pl -- this script performs the following:
+
+\- validates the tarchive
+
+\- converts DICOM data into MINC files
+
+\- automated protocol checks against the entries in the \`mri\_protocol\` table.
```
Comment (cmadjar) [inline docs/scripts_md/tarchiveLoader.md]: Maybe rephrase this section as below:

this script performs the following:
- validation of the tarchive
- conversion of DICOM datasets into MINC files
- automated protocol checks against entries in the mri_protocol and mri_protocol_checks tables

File: docs/scripts_md/tarchiveLoader.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+# SYNOPSIS
+
+perl tarchiveLoader.pl &lt;/path/to/DICOM-tarchive> \`\[options\]\`
+
+Available options are:
+
+\-profile                    : name of the config file in
```
Comment (cmadjar) [inline docs/scripts_md/tarchiveLoader.md]: "name" should be "Name" to stay consistent with other descriptions

File: docs/scripts_md/tarchiveLoader.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+\-newScanner                 : By default a new scanner will be registered if the
+                              data you upload requires it. You can risk turning
+                              it off
+
+\-keeptmp                    : Keep temp dir. Makes sense if have infinite space
+                              on your server
+
+\-xlog                       : Open an xterm with a tail on the current log file
+
+\-verbose                    : if set, be verbose
```
Comment (cmadjar) [inline docs/scripts_md/tarchiveLoader.md]: "if" should be "If" too

File: docs/scripts_md/tarchiveLoader.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+# SYNOPSIS
+
+perl tarchiveLoader.pl &lt;/path/to/DICOM-tarchive> \`\[options\]\`
+
+Available options are:
+
+\-profile                    : name of the config file in
+                              `../dicom-archive/.loris_mri`
+
+\-force                      : Forces the script to run even if the validation
```
Comment (cmadjar) [inline docs/scripts_md/tarchiveLoader.md]: "Forces" should be "Force" (no s)

File: docs/scripts_md/tarchiveLoader.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+                              has failed
+
+\-reckless                   : Upload data to database even if study protocol is
+                              not defined or violated
+
+\-globLocation               : Loosen the validity check of the tarchive allowing
+                              for the possibility that the tarchive was moved to
+                              a different directory
+
+\-noJIV                      : Prevents the JIVs from being created
```
Comment (cmadjar) [inline docs/scripts_md/tarchiveLoader.md]: "Prevents" should be "Prevent" (no s)

File: docs/scripts_md/tarchiveLoader.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+\-newScanner                 : By default a new scanner will be registered if the
+                              data you upload requires it. You can risk turning
+                              it off
+
+\-keeptmp                    : Keep temp dir. Makes sense if have infinite space
```
Comment (cmadjar) [inline docs/scripts_md/tarchiveLoader.md]: "Keep temp dir" should be "Keep temporary directory"

File: docs/scripts_md/tarchiveLoader.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+\-xlog                       : Open an xterm with a tail on the current log file
+
+\-verbose                    : if set, be verbose
+
+\-seriesuid                  : Only insert this SeriesUID
+
+\-acquisition\_protocol       : Suggest the acquisition protocol to use
+
+\-bypass\_extra\_file\_checks   : Bypasses extra\_file\_checks
```
Comment (cmadjar) [inline docs/scripts_md/tarchiveLoader.md]: "Bypasses" should be "Bypass"

File: docs/scripts_md/tarchiveLoader.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+# DESCRIPTION
+
+This script interacts with the NeuroDB database system. It will connect to/deal
+with/ modify contents of the following tables:
```
Comment (cmadjar) [inline docs/scripts_md/tarchiveLoader.md]: Would rephrase second sentence to:

"It will connect to fetch or modify contents of the following tables:"

File: docs/scripts_md/tarchiveLoader.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+\`session\`, \`parameter\_file\`, \`parameter\_type\`, \`parameter\_type\_category\`,
+\`files\`, \`mri\_staging\`, \`notification\_spool\`
+
+## Methods
+
+### logHeader()
```
Comment (cmadjar) [inline docs/scripts_md/tarchiveLoader.md]: add the input in the function name

File: docs/scripts_md/tarchiveLoader.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+Function that adds a header with relevant information to the log file
+
+INPUTS:
+ - $date       : Date and time of upload
+ - $tarchive   : Location of source data
+ - $TmpDir     : tmp dir location
```
Comment (cmadjar) [inline docs/scripts_md/tarchiveLoader.md]: Rephrase "tmp dir location" to "Location of the temporary directory"

File: docs/scripts_md/tarchiveLoader.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+
+None reported.
+
+# LICENSING
+
+License: GPLv3
+
+# AUTHORS
+
+J-Sebastian Muehlboeck based on Jonathan Harlap\\'s process\_uploads
```
Comment (cmadjar) [inline docs/scripts_md/tarchiveLoader.md]: Oops, missed one comment. The LORIS team should be added to the author list

**Comment by MounaSafiHarab** [comment]
> A few minor changes requested in the documentation.
> 
> Tested inserting a scan and it ran smoothly!
> 
> Let me know when you have done the changes and I'll approve the PR so we can merge it at the next imaging meeting

changes done. But yeah, let us first bring it up at the imaging meeting to see if anyone opposes adding the .pl extension to the file

File: docs/scripts_md/tarchiveLoader.md (hunk 9 above)
Comment (cmadjar) [reply docs/scripts_md/tarchiveLoader.md]: Just missing the inputs in the method name. Other than that all good

**Comment by MounaSafiHarab** [comment]
after discussion with @cmadjar 
we decided to keep the extension addition separate from the documentation (i.e. this) pull request.
Mouna to update the pull request accordingly

File: docs/scripts_md/tarchiveLoader.md (hunk 9 above)
Comment (MounaSafiHarab) [reply docs/scripts_md/tarchiveLoader.md]: @cmadjar 
this is done now!

**Comment by MounaSafiHarab** [comment]
@cmadjar 

did the change to revert the .pl addition. I created a redmine ticket to add the extension to all as a cleanup task here (RM#).
I also changed the tables to be in the C<>. Not sure I did that before for other scripts...
and finally added the LogHeader() inputs!

File: docs/scripts_md/tarchiveLoader.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+\- add a check for all programms that will be used (exists, but could
+  be better....)
+
+\- consider whether to add a check for registered protocols against the
+  tarchive db to save a few minutes of converting
+
+\- also add an option to make it interactively query user to learn new protocols
+
+\- this should be a separate program
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/scripts_md/tarchiveLoader.md]: @cmadjar 
good question; I am guessing that "this should be a separate program" refers to the step before of interactively querying the user.

to be honest, the whole TODO list is not clear to me; but I do not want to simply delete them since they were originally in the script. So this should be discussed as a group before this gets removed. what do u think?

[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/scripts_md/tarchiveLoader.md]: Sounds good!
